service/history/execution/context.go

```diff
@@ -1209,6 +1209,9 @@ func (c *contextImpl) getWorkflowExecutionWithRetry(
 	case *types.EntityNotExistsError:
 		// it is possible that workflow does not exists
 		return nil, err
+	case *persistence.ShardOwnershipLostError:
```

Comment:
Little surprised about this, @Shaddoll  do you think this being missing could have caused problems?

Reply:
Without this specific line change it would fall to default case below. Returned err type is not impacted. This switch case seems mostly about keeping error log noise under control

Reply:
true, I didn't read this carefully enough.